UnityProject/Assets/Scripts/Clothing/Eyewear/NightVisionGoggles.cs

```diff
@@ -1,57 +1,46 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
-using Mirror;
 using CameraEffects;
 
-public class NightVisionGoggles : NetworkBehaviour, IServerInventoryMove
+public class NightVisionGoggles : MonoBehaviour, IClientInventoryMove
 {
 	[SerializeField, Tooltip("How far the player will be able to see in the dark while he has the goggles on.")]
 	private Vector3 nightVisionVisibility;
 
 	[SerializeField, Tooltip("The default minimal visibility size.")]
 	private Vector3 defaultVisionVisibility;
 
-	
-	private void OnStartClient()
-	{
-		if(PlayerManager.LocalPlayerScript != null)
-		{
-			var item = PlayerManager.LocalPlayerScript.Equipment.GetClothingItem(NamedSlot.eyes).GameObjectReference;
-			if(item == gameObject) {enableEffect(true);}
-		}
-	}
 
-	public void OnInventoryMoveServer(InventoryMove info)
+
+	public void OnInventoryMoveClient(ClientInventoryMove info)
 		{
-			RegisterPlayer registerPlayer = info.ToRootPlayer;
+			var registerPlayer = PlayerManager.LocalPlayerScript;
 
 			if (info.ToSlot != null && info.ToSlot?.NamedSlot != null)
 			{
 
 				if (registerPlayer != null && info.ToSlot.NamedSlot == NamedSlot.eyes)
 				{
-					TargetOnWearing(registerPlayer.connectionToClient);
+					OnWearing();
 				}
 			}
 
 			if (info.FromSlot != null && info.FromSlot?.NamedSlot != null && info.ToSlot != null)
```

Comment:
if (info.FromSlot != null && info.FromSlot?.NamedSlot != null)

Reply:
as info.ToSlot != null is allowed to be null if we are removing it from the inventory directly